classes/pickup/pickup.gd

```diff
@@ -90,11 +85,19 @@ func _pickup_id_setup() -> void:
 
 
 func _pickup_sound():
-	if sfx != null and has_node("SFXCollect"):
-		# Find an object we know will survive this object's destruction.
-		var safe_sfx_root = $"/root/Main"
-		# Anchor the sound source to that, then play it.
-		ResidualSFX.new_from_existing(sfx, safe_sfx_root)
+	# Check to make sure the object is killable.
+	if respawn_seconds == 0.0:
+		# If the has_node() check did not exist, it could pass an arg previously freed.
+		if sfx != null and has_node("SFXCollect"):
+			# Find an object we know will survive this object's destruction.
+			var safe_sfx_root = $"/root/Main"
+			# Anchor the sound source to that, then play it.
+			ResidualSFX.new_from_existing(sfx, safe_sfx_root)
+		# Some pickups i.e. coins, have no sfx at all. We need to run both checks.
+		elif sfx != null and !has_node("SFXCollect"):
+			printerr("No SFXCollect Found :(")
```

Comment:
Don't...don't do this.

Save error messages for cases that _should not happen._ Items having no sfx node is intended behavior, and printing messages when intended behavior happens isn't usually helpful.

Reply:
This is indeed a case that should never happen. If this happens, that check means `sfx` should exist, but can't be found.

Reply:
I will improve my comments to be more clear.

Reply:
If I am understanding correctly, this error is printed when there is a sound to play, but no SFXCollect present to play it. In that situation, an error should be printed, no? So I don't see what @Koopa1018 is disputing.

Reply:
You are understanding correctly. I think the issue was my original comment, which was unclear on what the check actually does. This may have confused @Koopa1018 on the intent of the check.